Dockerfile.build

```diff
@@ -1,2 +1,2 @@
 FROM golang:1.8.3
-RUN curl -s -o - https://get.docker.com/builds/Linux/x86_64/docker-1.9.1 > /usr/local/bin/docker && chmod +x /usr/local/bin/docker
+RUN curl -s -o - https://get.docker.com/ | sed 's/docker-ce/docker-ce=17.12.0/' | sh
```

Comment:
I don't see how this is a maintainable solution. 🤷‍♂️

Reply:
How else do you want to see it? 🤔

Reply:
Using this method we are relying on output from an unknown shell script, an inline sed that will most certainly break down the road, and then blindly piping that into a shell to exec. Nope....

We'll need to find a way to request a specific version of docker either through apt or some other method.

Reply:
https://docs.docker.com/install/linux/docker-ce/debian/#install-using-the-repository

Reply:
we're doing that with the sed script, the package name is `docker-ce` on all platforms, you pin a version by defining it like: `docker-ce=17.12.0`

Reply:
I think you might misunderstand the issue I'm bringing up. Using `sed` to augment a stream from an uncontrollable source is not a maintainable solution i.e. we cannot guarantee the get.docker.com script won't change down the road. I'll open a new PR that uses `apt-get` to pin the version.